### PR TITLE
Clean wt_demog_2 and wt_practice surveys

### PR DESCRIPTION
**Pull request by as2388**
Will be easier to review once #12 is merged.

#12 is blocking on #11, which is blocking on https://github.com/AfricasVoices/CoreDataModules/pull/34

### CODE REVIEW

File: run_scripts/05_06_survey_auto_code.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+USER=$1
+DATA_ROOT=$2
+
+cd ../survey_auto_code
+
+mkdir -p "$DATA_ROOT/05 Auto-Coded"
+mkdir -p "$DATA_ROOT/06 Coda Files"
+
+# Auto-code wt_demog_1 and export to Coda. TODO: Auto-code the other surveys and export to Coda.
```
Comment (lukechurch) [inline run_scripts/05_06_survey_auto_code.sh]: TODO on separate line please

File: survey_auto_code/Pipfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[[source]]
+url = "https://pypi.python.org/simple"
+verify_ssl = true
+name = "pypi"
+
+[packages]
+CoreDataModules = {git = "https://www.github.com/AfricasVoices/CoreDataModules", editable = true, ref = "588c8bddfb2b6492655c67fcc9a22fd8a84d2dc0"}  # TODO: Update to a short ref when PR in CoreData is merged
```
Comment (lukechurch) [inline survey_auto_code/Pipfile]: TODO on separate line pls

File: survey_auto_code/survey_auto_code.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        "Idp (Text) - wt_demog_2": somali.DemographicCleaner.clean_yes_no,
+        "Origin_District (Text) - wt_demog_2": somali.DemographicCleaner.clean_somalia_district,
+
+        "Household_Sickness (Text) - wt_practice": somali.DemographicCleaner.clean_yes_no,
+        "Cholera_Vaccination (Text) - wt_practice": somali.DemographicCleaner.clean_yes_no,
+        "Trustworthy_Advisors (Text) - wt_practice": None
+    }
+
+    # Load data from JSON file
+    with open(demog_1_input_path, "r") as f:
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: Let's create a utility method for this. No urgency.

File: survey_auto_code/survey_auto_code.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                td.append_data(
+                    {"{}_clean".format(key): cleaner(td[key])},
+                    Metadata(user, Metadata.get_call_location(), time.time())
+                )
+
+    # Set missing entries in the raw data to 'NA'
+    for td in all_survey_data:
+        for key in cleaning_plan:
+            if key not in td:
+                td.append_data({key: "NA"}, Metadata(user, Metadata.get_call_location(), time.time()))
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: Please use a constant for NA

File: survey_auto_code/survey_auto_code.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    # Write json output
+    if os.path.dirname(json_output_path) is not "" and not os.path.exists(os.path.dirname(json_output_path)):
```
Comment (lukechurch) [inline survey_auto_code/survey_auto_code.py]: I thought we already had a helper for this?

Comment (as2388) [reply survey_auto_code/survey_auto_code.py]: We will once #12 is merged

**Pull request merged**